util/clusterauth/clusterauth.go

```diff
@@ -248,84 +247,36 @@ func GetServiceAccountBearerToken(clientset kubernetes.Interface, ns string, sa
 		return "", fmt.Errorf("failed to get token for serviceaccount %q: %w", sa, err)
 	}
 
-	return string(secret.Data["token"]), nil
+	return string(secret.Data[corev1.ServiceAccountTokenKey]), nil
 }
 
-// getOrCreateServiceAccountTokenSecret will check if a ServiceAccount
-// already has a kubernetes.io/service-account-token secret associated
-// with it or creates one if the ServiceAccount doesn't have one. This
-// was added to help add k8s v1.24+ clusters.
+// getOrCreateServiceAccountTokenSecret will create a
+// kubernetes.io/service-account-token secret associated with a
+// ServiceAccount named '<service account name>-long-lived-token', or
+// use the existing one with that name.
+// This was added to help add k8s v1.24+ clusters.
 func getOrCreateServiceAccountTokenSecret(clientset kubernetes.Interface, sa, ns string) (string, error) {
```

Comment:
I'd rather use longer names instead of `sa` and `ns`

Reply:
Fair enough, addressed.